### DISCUSSION TITLE
Syntax issues and questions

### DISCUSSION THREAD

**Discussion by levouh** [nvim-neorg/neorg#1029] [Syntax]
A few questions.

---

1. According to the specification, "wiki links" should link to an arbitrary heading in any file within the workspace.

If I have a structure like:

![image](https://github.com/nvim-neorg/neorg/assets/31262046/624fef94-2c03-4f4a-89ca-d5ee711ebc41)

and insert the following text into `other.norg`:

```
- A wiki link to {? Splunk}
```

then press `<CR>` or `gf` when hovered over the `{? Splunk}` text, it says the link is broken. The file `splunk.norg` contains:

```
* Splunk

```

Is this expected? Am I doing something wrong? For the record, the following works just fine:

```
- A file link to {:$/splunk:* Splunk}[Splunk]
```

---

2. I cannot get code block highlighting to work, despite having the TS parsers installed and _seemingly_ using the right syntax. E.g. a file called `other.norg` within the workspace has the following content:

```
- A file link to {:$/splunk:* Splunk}[Splunk]
- A wiki link to {? Splunk}

@code lua
local M = {}

return M
@end
```

and looks like:

![image](https://github.com/nvim-neorg/neorg/assets/31262046/f77429e1-e3ce-4b54-af9d-1c843f5608e1)

The syntax tree looks fine as well, noting the lack of any injected syntax trees though:

![image](https://github.com/nvim-neorg/neorg/assets/31262046/46d7b62d-1327-4d6b-8ccd-2bf99826d381)

The same thing happens even with `core.concealer` disabled. My configuration is as follows (using `lazy.nvim`):

```lua
return {
  "nvim-neorg/neorg",
  dependencies = {
    "nvim-lua/plenary.nvim",
    "nvim-treesitter/nvim-treesitter",
    "hrsh7th/nvim-cmp",
    "nvim-telescope/telescope.nvim",
    "nvim-neorg/neorg-telescope",
  },
  build = ":Neorg sync-parsers",
  config = function(_, opts)
    require("neorg").setup(opts)
    require("telescope").load_extension("neorg")
  end,
  opts = {
    load = {
      ["core.defaults"] = {},
      ["core.journal"] = { config = { strategy = "flat" } },
      ["core.qol.todo_items"] = { config = { create_todo_parents = true } },
      ["core.completion"] = { config = { engine = "nvim-cmp" } },
      ["core.ui.calendar"] = {},
      ["core.integrations.telescope"] = {},
      ["core.keybinds"] = {
        config = {
          neorg_leader = "<leader>n",
          hook = function(keybinds)
            keybinds.unmap("norg", "i", "<M-CR>")

            keybinds.map("norg", "n", "gj", "<Cmd>Neorg journal today<CR>")

            keybinds.remap_event("norg", "i", "<C-f>", "core.integrations.telescope.insert_link")

            keybinds.remap_event("norg", "n", "ge", "core.looking-glass.magnify-code-block")
            keybinds.remap_event("norg", "n", "<C-s>", "core.integrations.telescope.find_linkable")
          end,
        },
      },
      ["core.concealer"] = {
        config = {
          icons = {
            todo = {
              cancelled = { icon = "c" },
              done = { icon = "x" },
              on_hold = { icon = "h" },
              pending = { icon = "p" },
              recurring = { icon = "r" },
              uncertain = { icon = "?" },
              undone = { icon = " " },
              urgent = { icon = "!" },
            },
          },
        },
      },
      ["core.dirman"] = {
        config = {
          workspaces = {
            notes = "~/mnt/neorg",
          },
        },
      },
    },
  },
}
```

and I have ensured `:TSInstall lua` or `:TSUpdate lua` have been run. This happens with any language, and doing something like opening the code block in another buffer highlights the content just fine (as expected).

---

3. With the [absence of GTD](https://github.com/nvim-neorg/neorg/issues/695) in `1.0.0` (which by the way is _really_ hard to get any idea of how close it is to being re-implemented), I want to try to "port" over a system I've maintained myself. A lot of it I will add my own custom handling for, but a big part of my workflow is referencing items in files where they are not defined. Per the syntax, I can have a heading that is like:

```
#task.type work
* (# A| |@ 19th Aug 2023) Description of the task
```

which would specify that it is "undone", with priority `A`, and a date of `08/19/23`. It (and its children) are also tagged with a `task.type` value of `work`. Looking at the syntax tree, this is all parsed correctly and will be more than easy enough for me to work with.

However, where this tends to fall apart is with links. I cannot link to this task because the "todo matter" is considered part of the heading and is obviously subject to change - e.g. I want to change the status or date but still be able to link to the item.

I want to be able to have text in another file that does something like:

```
- This is a reference to {? Description of the Task}
```

and just navigate to the task as defined above directly.

Obviously something like:

```
#task.type work
* Description of the task
** (# A| |@ 19th Aug 2023)
```

would work as then I can still reference the task by its title, but having the "metadata" in a separate header like that doesn't seem like a great way to do things (and it is also harder to parse with a TS query).

Is there a better way to do this? "This" being "keep metadata about a heading but still be able to link to it as the metadata changes"?

**Comment by levouh** [comment]
@vhyrro not sure if you get pinged when discussions get created, and I apologize for doing so, but I'd be very interested in your insights in (1) and (3) noted above.

---

As for (2), I believe it is an "actual issue" so I will move it there as needed (see #1031)

**Reply by vhyrro** [reply]
Hey! yeah I didn't get pinged so thanks for doing that :)

1) Wiki links are unimplemented as I wait on the Treesitter V3 engine. With that I'll be more than happy to start work on that.
2) I believe this is because recently TS in neovim had a breaking change as to how they handle injections. I'll have to update my queries
3) That's weird, to my knowledge the metadata should be ignored and is never included in the link location. Does it successfully find your link if you have only one bit of metadata attached? Perhaps there's a logic bug where it doesn't skip all metadata during search 🤔. Feel free to open an issue!

Hope that helps :D

**Comment by max397574** [comment]
for 1 afaik wiki links are not implemented yet
